internal/pkg/cli/svc_init.go

```diff
@@ -604,24 +621,45 @@ func (o *initSvcOpts) askSvcPort() (err error) {
 		return nil
 	}
 
-	port, err := o.prompt.Get(
-		fmt.Sprintf(svcInitSvcPortPrompt, color.Emphasize("port")),
+	if o.wkldType == manifestinfo.RequestDrivenWebServiceType {
+		selectedPorts, err := o.prompt.Get(
+			fmt.Sprintf(svcInitSvcPortPrompt, color.Emphasize("port")),
+			svcInitSvcPortHelpPrompt,
+			validateSvcPort,
+			prompt.WithDefaultInput(defaultPort),
+			prompt.WithFinalMessage("Port:"),
+		)
+		if err != nil {
+			return fmt.Errorf("get port: %w", err)
+		}
+		if len(strings.Split(selectedPorts, ",")) > 1 {
+			return fmt.Errorf("App Runner Service doesn't expose multiple ports")
+		}
+		o.ports = make([]string, 1)
+		o.ports[0] = selectedPorts
+		return nil
+	}
+
+	selectedPorts, err := o.prompt.Get(
+		fmt.Sprintf(svcInitSvcPortPrompt, color.Emphasize("port(s)")),
 		svcInitSvcPortHelpPrompt,
 		validateSvcPort,
 		prompt.WithDefaultInput(defaultPort),
-		prompt.WithFinalMessage("Port:"),
+		prompt.WithFinalMessage("Port(s):"),
 	)
 	if err != nil {
 		return fmt.Errorf("get port: %w", err)
 	}
+	portList := strings.Split(selectedPorts, ",")
+	o.ports = make([]string, len(portList))
+	copy(o.ports, portList)
```

Comment:
Why do we need this?

Reply:
So if customers doesn't provide any port(s) in `--port` flag then we prompt for `Which port(s) do you want customer traffic sent to?`  and when they specify ports then we set those ports to `o.ports` field. Does that make sense?

Reply:
Why can't we just do `o.ports = portList`?